libmachine/util/util.go

```diff
@@ -30,3 +31,32 @@ func GetProxyURL(hostUrl string) (*url.URL, error) {
 	}
 	return proxy, nil
 }
+
+// SplitArgs splits the given args slice into two slices. The first slice will contain all arguments before the special
+// "--" argument separator, and the second will contain all arguments after it.
+// For example, if the arguments are
+//
+//	"./machine host1 host2 -- --driver-flag driver-arg"
+//
+// then the first slice of arguments will be
+//
+//	"./machine host1 host2"
+//
+// and the second slice of arguments will be
+//
+//	"--driver-flag driver-arg"
+func SplitArgs(args []string) ([]string, []string) {
+	leftArgs, rightArgs := make([]string, 0), make([]string, 0)
+	foundSep := false
+	for _, arg := range args {
+		if !foundSep && strings.TrimSpace(arg) == "--" {
+			foundSep = true
+		} else if !foundSep {
+			leftArgs = append(leftArgs, arg)
+		} else {
+			rightArgs = append(rightArgs, arg)
+		}
+	}
```

Comment:
Let's say, we hit the "--" for the first time, `if !foundSep` returns true so it appends "--" to the `leftArgs` and traverses to the next arg. As a result, `foundSep` never be true and all arg are append to `leftArgs`.